components/templates/src/filters.rs

```diff
@@ -78,6 +80,22 @@ pub fn base64_decode<S: BuildHasher>(
     Ok(to_value(as_str).unwrap())
 }
 
+pub fn replace_re(value: &Value, args: &HashMap<String, Value>) -> TeraResult<Value> {
+    let text = try_get_value!("replace_re", "value", String, value);
+    let pattern = match args.get("pattern") {
+        Some(val) => try_get_value!("replace_re", "pattern", String, val),
+        None => return Err(TeraError::msg("Filter `replace_re` expected an arg called `pattern`")),
+    };
+    let rep = match args.get("rep") {
+        Some(val) => try_get_value!("replace_re", "rep", String, val),
+        None => return Err(TeraError::msg("Filter `replace_re` expected an arg called `rep`")),
+    };
+    let pattern_re = Regex::new(&pattern)
```

Comment:
Can you cache the compiled regex? If someone puts that filter in their base template and they have a lot of pages, re-compiling it could take a significant amount of time wrt the build time.

Reply:
I'm rather new to rust (and zola) but I'll try and give it a shot.
Is there a place in zola which does caching already?

Reply:
I wasn't sure if I needed the same sync primitives for this filter as the one you have linked to, so I just included it. I hope something along those lines in commit 
185856d should work?